R/path.R

```diff
@@ -33,13 +32,40 @@ path_norm <- function(path) {
 path_expand <- function(path) {
   path <- enc2utf8(path)
 
-  enc2utf8(path.expand(path))
+  new_fs_filename(path.expand(path))
+}
+
+#' Tidy paths
+#'
+#' untidy paths are all different, tidy paths are all the same.
+#'
+#' Tidy paths always expand `~`, use `/` to delimit directories, never have
+#' multiple `/` or trailing `/` and have colourised output based on the file
+#' type.
+#'
+#' @return A fs_filename object
+#' @template fs
+#' @return a 
+#' @export
+path_tidy <- function(path) {
+  path <- path_expand(path)
+
+  # convert `\\` to `/`
+  path <- gsub("\\\\", "/", path)
+
+  # convert multiple // to single /
+  path <- gsub("//+", "/", path)
+
+  # Remove trailing / from paths
+  path <- sub("/$", "", path)
+
+  path
 }
 
 #' Provide the path to the users home directory
 #' @export
 path_home <- function() {
```

Comment:
Not really germane to this PR, but I think we should pass `...` onto to `path()`